frontend/app/routes/$lang+/_public+/apply+/$id+/date-of-birth.tsx

```diff
@@ -136,28 +147,35 @@ export async function action({ context: { session }, params, request }: ActionFu
     return json({ errors: parsedDataResult.error.format() });
   }
 
-  await applyRouteHelpers.saveState({ params, request, session, state: { dateOfBirth: parsedDataResult.data.dateOfBirth } });
+  await applyRouteHelpers.saveState({ params, request, session, state: parsedDataResult.data });
 
   const parseDateOfBirth = parse(parsedDataResult.data.dateOfBirth, 'yyyy-MM-dd', new Date());
   const age = differenceInYears(new Date(), parseDateOfBirth);
+  const childUnder18 = parsedDataResult.data.childUnder18;
+
+  if (age >= 65 && childUnder18 === 'yes') {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/applicant-information', params));
+  }
+
+  if (age >= 65 && childUnder18 === 'no') {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/apply-for-yourself', params));
+  }
 
   if (age >= 18 && age < 65) {
     return redirect(getPathById('$lang+/_public+/apply+/$id+/disability-tax-credit', params));
   }
 
-  if (age >= 16 && age < 18) {
+  if ((age === 16 || age === 17) && childUnder18 === 'yes') {
     return redirect(getPathById('$lang+/_public+/apply+/$id+/living-independently', params));
   }
 
-  if (age < 16) {
+  if (age < 16 && childUnder18 === 'yes') {
     return redirect(getPathById('$lang+/_public+/apply+/$id+/parent-or-guardian', params));
   }
 
   if (state.editMode) {
     return redirect(getPathById('$lang+/_public+/apply+/$id+/review-information', params));
   }
-
-  return redirect(getPathById('$lang+/_public+/apply+/$id+/applicant-information', params));
```

Comment:
We need a default redirect otherwise there might be scenario where the user won't be redirected. Review the conditions by reversing the conditions. Start by youger age first then move up to older age.